src/non-standalone/Dialog/ImageDialog.tsx

```diff
@@ -0,0 +1,247 @@
+	const classes = useStyles(props);
+	const {
+		setPrimaryLabel,
+		groupBoxLabel,
+		infoText,
+		readOnly,
+		downscale,
+		convertImagesTo,
+		uploadedImages,
+		onChange,
+		onPrimarySelected,
+		onUpdateImages,
+	} = props;
+	const [localuploadedImages, setImages] = useState<allImages[]>([
+		...uploadedImages,
+	]);
+	const closeDialog = () => popDialog();
+	const changeRef = useRef<HTMLInputElement>(null);
+	const processFile = useCallback(
+		async (file: File) => {
+			if (!changeRef.current) return;
+
+			const value = await processImage(file, convertImagesTo, downscale);
+			const newUploadedImages = [
+				...localuploadedImages,
+				{ src: value, setPrimary: false },
+			];
+			setImages(newUploadedImages);
+			// eslint-disable-next-line @typescript-eslint/unbound-method
+			const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+				window.HTMLInputElement.prototype,
+				"value"
+			)?.set;
+			nativeInputValueSetter?.call(changeRef.current, value);
+
+			const event = new Event("input", { bubbles: true });
+			changeRef.current.dispatchEvent(event);
+		},
+		[convertImagesTo, downscale, localuploadedImages, setImages]
+	);
+	// upload click handler
+	const handleUpload = useCallback(() => {
+		const elem = document.createElement("input");
+		elem.type = "file";
+		elem.accept = "image/*";
+		elem.multiple = false;
+		elem.addEventListener("change", () => {
+			const files = elem.files;
+			if (!files) return;
+			const file = files[0];
+			if (!file) return;
+			void processFile(file);
+		});
+		elem.click();
+	}, [processFile]);
+	const changePrimaryImage = useCallback(
+		(index) => {
+			const newUpdatedImages = localuploadedImages.map((ele, ind) => {
+				if (index === ind) {
+					ele.setPrimary = !ele.setPrimary;
+				} else {
+					ele.setPrimary = false;
+				}
+				return ele;
+			});
+			setImages(newUpdatedImages);
+			onPrimarySelected(newUpdatedImages);
+		},
+		[localuploadedImages, onPrimarySelected]
+	);
+	const removeImageHandler = useCallback(
+		(index) => {
+			localuploadedImages.splice(index, 1);
+			setImages([...localuploadedImages]);
+			onUpdateImages(localuploadedImages);
+		},
+		[localuploadedImages, setImages, onUpdateImages]
+	);
+	const handleDrop = useCallback(
+		async (evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+
+			await processFile(evt.dataTransfer?.files[0]);
+		},
+		[readOnly, processFile]
+	);
+
+	const handleDragOver = useCallback(
+		(evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+		},
+		[readOnly]
+	);
+	return (
+		<Dialog
+			onClose={closeDialog}
+			maxWidth="lg"
+			fullWidth
+			disableBackdropClick
+			open={true}
+		>
+			<MuiDialogTitle disableTypography>
+				<IconButton
+					aria-label="close"
+					className={classes.closeButton}
+					onClick={closeDialog}
+				>
+					<Close />
+				</IconButton>
+			</MuiDialogTitle>
+			<DialogContent>
+				<Grid item>
```

Comment:
move the DialogContent into it's own standalone component so it can be customized by library users